srml/support/src/additional_traits.rs

```diff
@@ -46,3 +46,23 @@ impl<T, U> ChargeFee<T> for DummyChargeFee<T, U> {
 	fn charge_fee(_: &T, _: Self::Amount) -> Result<(), &'static str> { Ok(()) }
 	fn refund_fee(_: &T, _: Self::Amount) -> Result<(), &'static str> { Ok(()) }
 }
+
+/// A type which can verify doughnuts for an inherent permission domain
+pub trait DoughnutVerifier<Doughnut> {
```

Comment:
Let's name this a `DoughnutDomainVerifier`.
Reasoning: `DoughnutVerifier` has another meaning within this context.

Reply:
I renamed this to `DispatchVerifier`